app/models/user.rb

```diff
@@ -0,0 +1,49 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :integer          not null, primary key
+#  email                  :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  reset_password_token   :string
+#  reset_password_sent_at :datetime
+#  remember_created_at    :datetime
+#  sign_in_count          :integer          default(0), not null
+#  current_sign_in_at     :datetime
+#  last_sign_in_at        :datetime
+#  current_sign_in_ip     :string
+#  last_sign_in_ip        :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  provider               :string
+#  uid                    :string
+#  role                   :string           default("coach"), not null
+#
+
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :recoverable, 
+    :rememberable, :trackable, :validatable, :omniauthable, :omniauth_providers => [:github]
+
+  ADMIN = "admin"
+  COACH = "coach"
+
+  def self.from_omniauth(auth)
+    email = auth.info.email.nil? ? auth.extra.raw_info.email : auth.info.email
+    login = auth.extra.raw_info.login
+    if !login.nil? and !email.nil?
```

Comment:
this block of code is ripe for pulling out as a separate method - we could also rewrite as `unless login.nil or email.nil?`

ah, I guess the method is not so huge because it's just block syntax on that first_or_create, but we could extract that as a separate method and self document like so:

unless login.nil or email.nil?
   find_or_create_user_from(login, email, auth)
end

just some thoughts ...